itest/rust/src/object_tests/property_test.rs

```diff
@@ -537,3 +537,85 @@ fn test_var_with_renamed_funcs() {
 
     obj.free();
 }
+
+#[derive(GodotClass)]
+#[class(base=Node)]
+struct SomeDuplicator {
+    #[var]
+    int_val: i32,
+
+    #[var]
+    optional_node: Option<Gd<Node>>,
+
+    #[export]
+    export_area: OnEditor<Gd<Area2D>>,
+}
+
+#[godot_api]
+impl INode for SomeDuplicator {
+    fn init(_base: Base<Node>) -> Self {
+        SomeDuplicator {
+            int_val: 0,
+            optional_node: None,
+            export_area: OnEditor::default(),
+        }
+    }
+}
+
+#[itest]
+fn test_some_duplicator() {
+    let mut obj = SomeDuplicator::new_alloc();
+    assert_eq!(obj.bind().int_val, 0);
+    assert_eq!(obj.bind().optional_node, None);
+
+    obj.bind_mut().int_val = 5;
+    assert_eq!(obj.bind().int_val, 5);
+
+    let some_node = Some(Node::new_alloc());
+    let some_id = some_node.as_ref().unwrap().instance_id();
+    obj.bind_mut().optional_node = some_node.clone();
+    let obj_id = obj.bind().optional_node.as_ref().unwrap().instance_id();
+    assert_eq!(obj_id, some_id);
```

Comment:
You don't need to _declare_ `some_node` as `Option`, just wrap it in `Some()` during the assignment. This also gets rid of `unwrap`:
```suggestion
    let some_node = Node::new_alloc();
    let some_id = some_node.instance_id();
    obj.bind_mut().optional_node = Some(some_node.clone());
```

The last two lines are not necessary:
```rs
    let obj_id = obj.bind().optional_node.as_ref().unwrap().instance_id();
    assert_eq!(obj_id, some_id);
```
Because also here, you're just testing whether `bind()` returns the value you just set, and whether `Clone` works and returns the same instance ID. Which we know is already working.